src/main/java/com/wiley/autotest/selenium/elements/upgrade/NullTeasyElement.java

```diff
@@ -0,0 +1,237 @@
+package com.wiley.autotest.selenium.elements.upgrade;
+
+import com.wiley.autotest.selenium.context.SearchStrategy;
+import com.wiley.autotest.selenium.elements.upgrade.v3.*;
+import org.openqa.selenium.*;
+import org.openqa.selenium.interactions.internal.Coordinates;
+import org.openqa.selenium.internal.Locatable;
+
+import java.util.List;
+
+
+/**
+ * Represents element that is absent (not found)
+ */
+public class NullTeasyElement implements TeasyElement, Locatable {
+
+    private TeasyElementData elementData;
+
+    public NullTeasyElement(TeasyElementData elementData) {
+        this.elementData = elementData;
+    }
+
+    @Override
+    public Should should() {
+        return new NullShouldImmediately(this);
+    }
+
+    @Override
+    public Should should(SearchStrategy strategy) {
+        return new NullShould(elementData, strategy);
+    }
+
+    @Override
+    public ElementWaitFor waitFor() {
+        return new NullElementWaitForImmediately(this);
+    }
+
+    @Override
+    public ElementWaitFor waitFor(SearchStrategy strategy) {
+        return new NullElementWaitFor(elementData, strategy);
+    }
+
+    /*
+      All other methods of TeasyElement should throw an exception because it's not possible to
+      interact with element that does not exist
+     */
+
+    @Override
+    public <X> X getScreenshotAs(OutputType<X> target) throws WebDriverException {
```

Comment:
Is it necessary here?
throws WebDriverException

Reply:
@FedVl seems like it is not. Removed from here and from BaseTeasyElement